providers/embedded_ansible.md

```diff
@@ -0,0 +1,231 @@
+
+```
+psql -d postgres -c 'DROP DATABASE awx'
+psql -d postgres -c 'DROP ROLE awx'
+bin/rake evm:db:reset
+bin/rake db:seed
+```
+
+
+If you had previously added an embedded ansible using the [old way](http://talk.manageiq.org/t/howto-setup-embedded-ansible/2291/5?u=himdel), you'll need to clean up the provider (in Rails console):
+
+```
+ManageIQ::Providers::EmbeddedAnsible::Provider.first.destroy!
+```
+
+
+In both cases, you may also need to clean up the old authentications (in Rails console):
+
+```
+db = MiqDatabase.first
+db.authentication_type('ansible_secret_key').delete    # db.ansible_secret_key.delete
+db.ansible_rabbitmq_authentication.delete
+db.ansible_admin_authentication.delete
+db.ansible_database_authentication.delete
+```
+
+
+### Procfiles
+
+Under your `manageiq/` directory, create these 2 files:
+
+`Procfile.ansible`:
+
+```
+ansible: ruby lib/workers/bin/run_single_worker.rb EmbeddedAnsibleWorker
+```
+
+`Procfile.workers`:
+
+```
+generic:                  ruby lib/workers/bin/run_single_worker.rb MiqGenericWorker
+embedded_ansible_refresh: ruby lib/workers/bin/run_single_worker.rb -e 123 ManageIQ::Providers::EmbeddedAnsible::AutomationManager::RefreshWorker
+embedded_ansible_event:   ruby lib/workers/bin/run_single_worker.rb -e 123 ManageIQ::Providers::EmbeddedAnsible::AutomationManager::EventCatcher
+```
+
+In the second file, you'll need to replace that 123 with the id of the newly created **manager** instance.
+
+
+### Setting it up
+
+  * configure your server to enable the ansible role (from Rails console):
+
+```
+server = MiqServer.my_server(true)
+server.role = "embedded_ansible,ems_inventory,ems_operations,event"
+server.activate_roles(%w(embedded_ansible ems_inventory ems_operations event))
+server.save!
+```
+
+   * run rails: `bin/rails s`
+
+   * run the worker that will download and set up the container: `foreman start -f Procfile.ansible`
+
+   * grab a coffee or two - you can watch the progress by watching:
+      * authentication errors, docker problems: `tail -f managiq/log/evm.log`
+      * running containers: `docker ps`
+      * container logs: `docker logs -f awx_web`
+      * seeing awx initial upgrade progress `localhost:54321`
+
+   * once everything suceeded, you should see `Finished starting embedded ansible service.` in `evm.log`
+
+   * if you got that far, AWX is running and ManageIQ has an EmbeddedAnsible provider instance
+
+   * you need to edit `Procfile.workers`, to replace that `123` with the actual id of the new manager (not provider) instance:
+
+```
+ManageIQ::Providers::EmbeddedAnsible::Provider.first.managers.first.id
+```
+
+   * run `foreman start -f Procfile.workers`
+
+   * try adding a Repository in ManageIQ (Automate > Ansible > Repositories) :)
+
+
+If you're on MacOSX, you will also need to run these first:
+
+```
+# redirect local 54321 to docker-machine - otherwise, localhost:54321 doesn't work
+docker-machine ssh default -L 54321:127.0.0.1:54321
```

Comment:
Wouldn’t it be better to set up a proper port forwarding in VirtualBox? This way it is not necessary to keep the SSH session open. It is a less hacky solution too.

See https://stackoverflow.com/questions/32174560/port-forwarding-in-docker-machine

You’d need to run something like this

```
VBoxManage modifyvm "default" --natpf1 "awx,tcp,127.0.0.1,54321,,54321"
```

or you can set it up in the VirtualBox GUI.

Reply:
Sounds better, how would you do that so that it only affects the docker machine and not other VirtualBox machines?

Reply:
The setting is machine-specific. The `default` there in the command is a name of the Docker Machine.

Reply:
mentioned as an alternative